File: TSLMS_Backend/HolidayService/Models/Holiday.cs
namespace HolidayService.Models
{
    public class Holiday
    {
        public Guid Id { get; set; }
        public string Name { get; set; } = string.Empty;
        public DateOnly HolidayDate { get; set; }
        public string? Description { get; set; }
        public DateTime CreatedAtUtc { get; set; }
        public DateTime UpdatedAtUtc { get; set; }
    }
}

File: timesheet-leave-management-frontend/src/app/models/holiday.ts
export interface Holiday {
  id: string;
  name: string;
  holidayDate: string;
  description: string | null;
  year: number;
  dayOfWeek: string;
  createdAtUtc: string;
  updatedAtUtc: string;
}

export interface CreateHolidayDto {
  name: string;
  holidayDate: string;
  description?: string;
}

export interface UpdateHolidayDto extends CreateHolidayDto {}

export interface CopyHolidayYearDto {
  sourceYear: number;
  targetYear: number;
  skipExistingDates: boolean;
}

export interface HolidayCheck {
  date: string;
  isHoliday: boolean;
  holidayId: string | null;
  holidayName: string | null;
}
